### PR TITLE
refactor!: use packageName and not depName for datasource lookups

### PR DESCRIPTION
**Pull request by rarkins**
Refactors datasources + lookup to use `packageName` primarily and not `depName`.

BREAKING CHANGE: Anyone using renovate as a library and calling getPkgReleases must supply packageName and not depName

### CODE REVIEW

File: lib/workers/repository/process/fetch.ts [hunk 1]
```diff
@@ -28,7 +28,8 @@ async function fetchDepUpdates(
   if (is.string(dep.depName)) {
     dep.depName = dep.depName.trim();
   }
-  if (!is.nonEmptyString(dep.depName)) {
+  dep.packageName ??= dep.depName;
+  if (!is.nonEmptyString(dep.packageName)) {
```
Comment (rarkins) [inline lib/workers/repository/process/fetch.ts:32]: The `depConfig as LookupUpdateConfig` on line 60 meant our type checking didn't catch/enforce this

File: lib/modules/datasource/index.ts [hunk 2]
```diff
@@ -437,7 +441,7 @@ function getDigestConfig(
 ): DigestConfig {
   const { currentValue, currentDigest } = config;
   const packageName =
-    config.replacementName ?? config.packageName ?? config.depName;
+    config.replacementName ?? config.packageName ?? config.packageName;
```
Comment (viceice) [inline lib/modules/datasource/index.ts:444]:
```suggestion
    config.replacementName ?? config.packageName;
```
duplicate 😉

File: lib/modules/datasource/index.ts [hunk 3]
```diff
@@ -342,7 +346,7 @@ export async function getPkgReleases(
     logger.warn('No datasource found');
     return null;
   }
-  const packageName = config.packageName ?? config.depName;
+  const packageName = config.packageName ?? config.packageName;
```
Comment (viceice) [inline lib/modules/datasource/index.ts:349]:
```suggestion
  const packageName = config.packageName;
```

File: lib/modules/datasource/docker/index.ts [hunk 4]
```diff
@@ -347,14 +347,14 @@ export function isECRMaxResultsError(err: HttpError): boolean {
 }
 
 const defaultConfig = {
-  commitMessageTopic: '{{{depName}}} Docker tag',
+  commitMessageTopic: '{{{packageName}}} Docker tag',
```
Comment (viceice) [inline lib/modules/datasource/docker/index.ts:350]: are you sure about this change?

File: lib/modules/datasource/docker/index.ts [hunk 5]
```diff
@@ -347,14 +347,14 @@ export function isECRMaxResultsError(err: HttpError): boolean {
...
   commitMessageExtra:
     'to {{#if isPinDigest}}{{{newDigestShort}}}{{else}}{{#if isMajor}}{{{prettyNewMajor}}}{{else}}{{{prettyNewVersion}}}{{/if}}{{/if}}',
   digest: {
     branchTopic: '{{{depNameSanitized}}}-{{{currentValue}}}',
     commitMessageExtra: 'to {{newDigestShort}}',
     commitMessageTopic:
-      '{{{depName}}}{{#if currentValue}}:{{{currentValue}}}{{/if}} Docker digest',
+      '{{{packageName}}}{{#if currentValue}}:{{{currentValue}}}{{/if}} Docker digest',
```
Comment (viceice) [inline lib/modules/datasource/docker/index.ts:357]: see above

File: lib/modules/datasource/docker/index.ts (hunk 4 above)
Comment (rarkins) [inline lib/modules/datasource/docker/index.ts:350]:
```suggestion
  commitMessageTopic: '{{{depName}}} Docker tag',
```

Comment (rarkins) [reply lib/modules/datasource/docker/index.ts:350]: Accidental bulk replace

File: lib/modules/datasource/helm/index.ts [hunk 6]
```diff
@@ -20,7 +20,7 @@ export class HelmDatasource extends Datasource {
   override readonly defaultRegistryUrls = ['https://charts.helm.sh/stable'];
 
   override readonly defaultConfig = {
-    commitMessageTopic: 'Helm release {{depName}}',
+    commitMessageTopic: 'Helm release {{packageName}}',
```
Comment (viceice) [inline lib/modules/datasource/helm/index.ts:23]: this should also reverted?

File: lib/workers/repository/process/fetch.ts (hunk 1 above)
Comment (viceice) [reply lib/workers/repository/process/fetch.ts:32]: yes, there is a lot room for type improvement

File: lib/modules/datasource/helm/index.ts (hunk 6 above)
Comment (rarkins) [inline lib/modules/datasource/helm/index.ts:23]:
```suggestion
    commitMessageTopic: 'Helm release {{depName}}',
```

File: lib/modules/manager/bazel/parser.spec.ts [hunk 7]
```diff
@@ -228,6 +228,12 @@ describe('modules/manager/bazel/parser', () => {
             group = "com.example2",
             artifact = "bar",
             version = "2.2.2",
+          ),
+          maven.artifact(
```
Comment (viceice) [inline lib/modules/manager/bazel/parser.spec.ts:232]: looks like a bad merge. 🤔

File: lib/modules/manager/swift/extract.ts [hunk 8]
```diff
@@ -119,6 +119,7 @@ function getMatch(str: string, state: string | null): MatchResult | null {
 }
 
 function getDepName(url: string | null): string | null {
+  // istanbul ignore if
```
Comment (viceice) [inline lib/modules/manager/swift/extract.ts:122]: why this is no longer covered? 🤔

**Review by viceice — COMMENTED** [review]
one last thing, otherwise LGTM

**Pull request merged**